test/extensions/filters/network/ratelimit/ratelimit_test.cc

```diff
@@ -53,6 +55,8 @@ class RateLimitFilterTest : public testing::Test {
     Json::ObjectSharedPtr json_config = Json::Factory::loadFromString(json);
     envoy::config::filter::network::rate_limit::v2::RateLimit proto_config{};
     Envoy::Config::FilterJson::translateTcpRateLimitFilter(*json_config, proto_config);
+    // TODO(ramaraochavali): move the config to yaml and use a different config for failure_mode.
```

Comment:
I would go ahead and do this, and change these tests to use a SetUpTest(const std::string& yaml) setup function. Prevents the subclass addition here, as you point out in the TODO.